dch/public/app/components/watch-form/watch-form.js

```diff
@@ -0,0 +1,205 @@
+(function() {
+
+	'use strict';
+
+  /**
+   * @desc - Directive used to create the watch form query. Posts
+   *         user selected information to the DCH server.
+   *
+   * @example <watch-form></watch-form>
+   */
+   angular
+    .module('dcgui.components')
+    .directive('watchForm', watchForm)
+    // This filter is temporary. Will later use the API to directly
+    // retrieve the WatchIDs. Right now it simply removes duplicates
+    // of WatchIDs when populating the list of WatchIDs.
+    .filter('unique', function(){
+    	return function(collection, keyname) {
+    		var output = [],
+    		keys = [];
+
+    		angular.forEach(collection, function(item) {
+    			var key = item[keyname];
+    			if(keys.indexOf(key) === -1) {
+    				keys.push(key);
+    				output.push(item);
+    		}
+    	});
+    	return output;
+    	}
+    });
+
+    function watchForm() {
+
+    	var directive = {
+    		scope: {
+    			watches: '='
+    		},
+    		templateUrl: '/app/components/watch-form/watch-form.html',
+    		controller: WatchFormController,
+    		controllerAs: 'watchFormCtrl',
+    		restrict: 'E',
+    		bindToController: true
+    	};
+
+    	return directive;
+
+    }
+
+    WatchFormController.$inject = ['$http', '$scope'];
+
+    function WatchFormController($http, $scope) {
+
+        $scope.watchSelected = false;
+        $scope.timeSelected = false;
+        $scope.times = [];
+
+        // This will store the user selected information.
+    	$scope.selectedWatch =  {
+    		id: '',
+    		startTime: '',
+            interval: '',
+    	}
+
+    	var vm = this;	
+
+    	var NO_DEVICE_DATA_DISPLAY = 'No data';
+    	var GET_DATA_QUERY_PATH = '/get_data/';
+
+    	vm.watchFormData = NO_DEVICE_DATA_DISPLAY;
+
+        // This watches for if a watch has been selected by a user.
+        $scope.$watch('selectedWatch.id', function () {
+            if ($scope.selectedWatch.id === '') {
```

Comment:
Will probably let this slide for now but beware of the ```$watch``` function. It can be expensive to use and it's a pain in the ass to test. In most cases you're better off without it.

E.g. in this case you could include the ```ng-show``` attribute like so:

``` ng-show="selectedWatch.id" ```